### PR TITLE
Add documentation to print compiler logs

### PR DESCRIPTION
**Pull request by ksolana**
Addresses: #142 for now. We don't strictly need a debug level at the moment.

### CODE REVIEW

**Review by dmakarov — APPROVED** [review]
We could also output debug prints in `debug!` (or maybe `dbg!` macro is already bound to logging on debug level). Then running the compiler with something like `RUST_LOG=debug` environment setting would print the debug messages.

**Comment by ksolana** [comment]
> We could also output debug prints in `debug!` (or maybe `dbg!` macro is already bound to logging on debug level). Then running the compiler with something like `RUST_LOG=debug` environment setting would print the debug messages.

Thanks for the pointer. looking at the different debug macros in [env_logger](https://docs.rs/env_logger/latest/env_logger/), it seems we should have a similar level for `--debug`. this makes it easier for controlling debug output.

File: language/tools/move-mv-llvm-compiler/src/stackless/translate.rs [hunk 1]
```diff
@@ -121,6 +121,7 @@ impl<'up> GlobalContext<'up> {
         &'this self,
         id: mm::ModuleId,
         dot_info: &'this String,
+        log_level: String,
```
Comment (nvjle) [inline language/tools/move-mv-llvm-compiler/src/stackless/translate.rs]: It is orthogonal to this patch, but now that we are adding more command line options, it isn't going to be scalable (or at least desirable) to keep adding each one to the contexts. It probably makes sense now to make `main.rs:Args` publicly/globally available. This is what `move-prover` does as well (`move-prover/src/cli.rs`). I like the way they isolate the args struct and command line processing to `cli.rs`. Also note that they support logging there.

Comment (ksolana) [reply language/tools/move-mv-llvm-compiler/src/stackless/translate.rs]: Yeah that makes more sense. I'll create a sub task as part of #138

**Comment by brson** [comment]
With the env_logger and log crates there's no need to pass a log level into the program at all. It's conventional to use those two crates together and have the caller set RUST_LOG to get logs. In the code, authors use `info!`, `error!`, `warn!` to log at different levels.

**Comment by ksolana** [comment]
> With the env_logger and log crates there's no need to pass a log level into the program at all. It's conventional to use those two crates together and have the caller set RUST_LOG to get logs. In the code, authors use `info!`, `error!`, `warn!` to log at different levels.

agreed, at this point there is no strong need to have a separate flag. if we need this in future we can add. I'll abandon the flag and add documentation.

**Pull request merged**